mrpt_bridge/src/pose.cpp

```diff
@@ -170,22 +170,25 @@ geometry_msgs::PoseWithCovariance& mrpt_bridge::convert(const mrpt::poses::CPose
   // Old comment: "MRPT uses non-fixed axis for 6x6 covariance: should use a transform Jacobian here!"
   //           JL ==> Nope! non-fixed z-y-x equals fixed x-y-z rotations.
 
-  for (int i = 0; i < 6; i++) {
-    for (int j = 0; j < 6; j++) {
-      double cov_val;
-      int ros_i = i;
-      int ros_j = j;
-      if (i > 2 || j > 2)
-        cov_val = 0;
-      else
-      {
-        ros_i = i == 2 ? 5 : i;
-        ros_j = j == 2 ? 5 : j;
-        cov_val = _src.cov(i, j);
-      }
-      _des.covariance[ros_i * 6 + ros_j] = cov_val;
```

Comment:
When i=5, j=5 (thus ros_i = 5, ros_j = 5) _des.covariance[5*6+5] is always 0 (since i > 2 and j > 2) even if it was modified prior to this, at the point where ros_i = i == 2 ? 5 : i;...